<launch>
    <!-- STRETCH DRIVER -->
    <param name="/stretch_driver/broadcast_odom_tf" type="bool" value="true"/>
    <param name="/stretch_driver/mode" type="string" value="navigation" />
    <include file="$(find stretch_core)/launch/stretch_driver.launch" pass_all_args="true"/>
    <include file="$(find stretch_core)/launch/d435i_high_resolution.launch"/>
    <include file="$(find stretch_core)/launch/stretch_aruco.launch" />

    <arg name="map_yaml" default="/home/team1/catkin_ws/src/team1_teleop/maps/new_room.yaml" doc="filepath to previously captured map (required)" />
    <arg name="rviz"     default="false"                                                                          doc="whether to show Rviz" />

  
    <!-- CENTERED BASE LINK -->
    <node name="centered_base_link_tf_publisher" pkg="tf" type="static_transform_publisher" args="-0.1 0 0 0 0 0 1 /base_link /centered_base_link 100" />

    <!-- WALKER FRAME-->
    <node name="walker_center_transformer" pkg="team1_teleop" type="walker_center_transformer"/>

    <!-- LASER RANGE FINDER -->
    <include file="$(find stretch_core)/launch/rplidar.launch" />

    <!-- MAP SERVER -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_yaml)" />

    <!-- LOCALIZATION -->
    <include file="$(find amcl)/examples/amcl_diff.launch" />

    <!-- NAVIGATION -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find stretch_navigation)/config/common_costmap_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find stretch_navigation)/config/common_costmap_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find stretch_navigation)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find stretch_navigation)/config/global_costmap_params_withmap.yaml" command="load" />
        <rosparam file="$(find stretch_navigation)/config/base_local_planner_params.yaml" command="load" />

        <remap from="/cmd_vel" to="/stretch/cmd_vel" />
    </node>

    <!-- Websocket -->
    <!-- <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"/> -->

    <!-- VISUALIZE -->
    <node name="rviz" pkg="rviz" type="rviz" output="log" args="-d $(find stretch_navigation)/rviz/navigation.rviz" if="$(arg rviz)" />

    <!-- PPL DETECT -->
    <arg name="debug_directory" value="$(env HELLO_FLEET_PATH)/debug/"/>

    <!-- STRETCH DRIVER -->
    <param name="/stretch_driver/fail_out_of_range_goal" type="bool" value="false"/>
    <!-- -->

    <!-- MAPPING -->
    <node name="funmap" pkg="stretch_funmap" type="funmap" output="screen" >
        <param name="debug_directory" type="string" value="$(arg debug_directory)"/>
    </node>
    <!-- -->

    <!-- LASER SCAN MATCHER FOR ODOMETRY -->
    <include file="$(find stretch_core)/launch/stretch_scan_matcher.launch" />
    <!-- -->

    <!-- HANDOVER POSITION DETECTOR -->
    <node name="detect_handover_position" pkg="stretch_deep_perception" type="detect_nearest_mouth.py" output="screen" />
    <!-- -->

</launch>